backend/src/app.controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { Node } from './entities/Node';
+import { Edge } from './entities/Edge';
+import { DetailedEdge } from './entities/DetailedEdge';
+import { ParseIntArrayPipe } from './pipes/ParseIntArrayPipe';
+import { AppService } from './app.service';
+
+/**
+ * Main App Controller.
+ * ATM there are no only a few endpoints, so only one controller is used.
+ */
+@Controller()
+export class AppController {
+  constructor(private readonly appService: AppService) {}
+
+  /**
+   * Returns a list the ids of all nodes
+   */
+  @Get('getAllNodes')
+  async getAllNodes(): Promise<number[]> {
```

Comment:
I propose to combine the methods `getAllNodes()` and `getAllEdges()` into a single method, that maybe named, f.e. `query()`.
This is benefical, as it truly represents the intend of the API and also serves the UI (frontend) with a dedicated model, if it returns a value of the to be defined type
```
interface QueryResult {
  nodes: number[],
  edges: EdgeDescriptor[]
}
```